### PR TITLE
Quick claim button

### PR DESCRIPTION
**Pull request by UmairMukhtar**

### CODE REVIEW

**Comment by UmairMukhtar** [comment]
https://github.com/eLearningDAO/POCRE/issues/277

**Comment by AltiMario** [comment]
image

**Comment by UmairMukhtar** [comment]
Creation id will now include base url from the window
Screenshot from 2023-02-18 19-22-15

**Review by AltiMario — APPROVED** [review]

**Comment by UmairMukhtar** [comment]
Fixed the material disappearing issue

Uploading after publishing the material disappear issue fixed.mp4…

**Comment by UmairMukhtar** [comment]
Not Claimable Creation
Screenshot from 2023-02-18 21-11-21

Claimable Creation
Screenshot from 2023-02-18 21-12-44

File: app/web-frontend/src/routes/index.jsx [hunk 1]
```diff
@@ -140,6 +140,16 @@ function AppRoutes() {
             </Protected>
           )}
         />
+        <Route
+          path="/litigations/create/:creationId/:materialId"
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/routes/index.jsx]: done

File: app/web-frontend/src/components/cards/MaterialCard/index.jsx [hunk 2]
```diff
@@ -153,7 +160,20 @@ function MaterialCard({
               />
             )
           }
-
+          {
+            isClaimable && userProfileId !== user && (
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/components/cards/MaterialCard/index.jsx]: done

File: app/web-frontend/src/pages/Litigations/Create/index.jsx [hunk 3]
```diff
@@ -1,8 +1,10 @@
+import { useParams } from 'react-router-dom';
 import LitigationForm from '../common/form';
 
 function CreateLitigation() {
+  const { creationId, materialId } = useParams();
   return (
-    <LitigationForm />
+    <LitigationForm creationId={creationId} materialId={materialId} />
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Litigations/Create/index.jsx]: done

File: app/web-frontend/src/pages/Litigations/common/form/index.jsx [hunk 4]
```diff
@@ -62,6 +66,8 @@ function LitigationForm({ id = null, onLitigationFetch = () => {} }) {
           description: litigation?.litigation_description,
           creation: litigation?.creation,
         }}
+        parameterCreationId={autofillClaim.creationId}
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Litigations/common/form/index.jsx]: done

File: app/web-frontend/src/routes/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/routes/index.jsx]: done

File: app/web-frontend/src/pages/Litigations/common/form/Steps/One.jsx [hunk 5]
```diff
@@ -35,6 +37,49 @@ export default function StepOne({
...
+    } finally {
+      setLoading(false);
+    }
+
+    setError(null);
+  };
+
+  useEffect(() => {
+    if (parameterCreationId) {
+      document.getElementsByName('creationLink')[0].value = `${window.location.origin}/creations/${parameterCreationId}`;
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Litigations/common/form/Steps/One.jsx]: done

**Comment by UmairMukhtar** [comment]
A litigation is successfully finalised 

https://user-images.githubusercontent.com/34018917/221099701-beebdc2d-688b-408e-8d45-184a8a49dd6d.mp4

**Comment by UmairMukhtar** [comment]
A litigation is successfully created in draft

https://user-images.githubusercontent.com/34018917/221100470-541b5089-1243-4e5b-9783-f1e9f66bdd53.mp4

**Comment by UmairMukhtar** [comment]
A litigation in draft is successfully updated

https://user-images.githubusercontent.com/34018917/221100762-cdd46521-d92b-47af-921d-459b0a78d7a1.mp4

**Comment by UmairMukhtar** [comment]
A litigation in draft is successfully finalised

https://user-images.githubusercontent.com/34018917/221101026-99e164d4-cef3-412d-8b62-1a5e44ece9fe.mp4

**Comment by UmairMukhtar** [comment]
Issue Fixed
https://github.com/eLearningDAO/POCRE/issues/277

**Pull request merged**